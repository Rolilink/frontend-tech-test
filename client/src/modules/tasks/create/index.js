import connectToState from './providers/CreateTask';
import CreateTaskComponent from './components/CreateTask';

export default connectToState(CreateTaskComponent);
