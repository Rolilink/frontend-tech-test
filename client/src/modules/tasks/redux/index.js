import * as actionTypes from './actionTypes';
import * as factories from './factories';
import * as selectors from './selectors';
import * as utils from './utils';

import { fetchTasks, createTask } from './actions';

import reducerModule from './reducers';

const reducer = {
  name: 'tasks', reducer: reducerModule,
};

export {
  fetchTasks,
  createTask,
  actionTypes,
  factories,
  selectors,
  reducer,
  utils,
};
