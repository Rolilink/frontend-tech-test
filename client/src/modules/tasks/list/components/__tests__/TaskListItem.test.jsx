import React from 'react';
import { shallow } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import TaskListItem from '../TaskListItem';
import { taskFactory } from '../../../../../__tests__/testUtils/tasks';

expect.addSnapshotSerializer(serializer);

const task = taskFactory({
  id: 1,
  title: 'Task Title',
  description: 'Task Description',
});

describe('Tasks.List.TaskListItem', () => {
  it('should render', () => (
    expect(shallow(<TaskListItem task={task} />)).toMatchSnapshot()
  ));
});
