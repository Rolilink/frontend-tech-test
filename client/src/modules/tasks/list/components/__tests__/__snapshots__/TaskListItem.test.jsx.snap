// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks.List.TaskListItem should render 1`] = `
<ListGroupItem
  bsClass="list-group-item"
  className="TaskListItem TaskListItem-1 "
  listItem={false}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      className="TaskListItem-CheckBoxContainer"
      componentClass="div"
      md={2}
      sm={2}
    >
      <Checkbox
        bsClass="checkbox"
        disabled={false}
        inline={false}
        title=""
      />
    </Col>
    <Col
      bsClass="col"
      className="TaskListItem-TitleAndDescription"
      componentClass="div"
      md={8}
      sm={8}
    >
      <Row
        bsClass="row"
        className="TaskListItem-TitleContainer"
        componentClass="div"
      >
        <Title
          className="TaskListItem-Title"
        >
          Task Title
        </Title>
      </Row>
      <Row
        bsClass="row"
        className="TaskListItem-DescriptionContainer"
        componentClass="div"
      >
        <Description
          className="TaskListItem-Description"
        >
          Task Description
        </Description>
      </Row>
    </Col>
    <Col
      bsClass="col"
      className="TaskListItem-CheckBoxContainer"
      componentClass="div"
      md={2}
      sm={2}
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="warning"
        disabled={false}
      >
        Edit
      </Button>
    </Col>
  </Row>
</ListGroupItem>
`;
